src/constructs/core/ssm.ts

```diff
@@ -0,0 +1,90 @@
+import { readFileSync } from "fs";
+import { join } from "path";
+import type { IGrantable, IPrincipal } from "@aws-cdk/aws-iam";
+import { Policy, PolicyStatement } from "@aws-cdk/aws-iam";
+import { Code, Runtime, SingletonFunction } from "@aws-cdk/aws-lambda";
+import type { Reference } from "@aws-cdk/core";
+import { Construct, CustomResource, Duration } from "@aws-cdk/core";
+import { AwsCustomResourcePolicy } from "@aws-cdk/custom-resources";
+import type { CustomResourceGetParameterProps } from "./custom-resources/interfaces";
+import type { GuStack } from "./stack";
+
+export interface GuSSMParameterProps {
+  secure?: boolean;
+  /*
+   * Assumes the path of `/${STAGE}/${STACK}/${APP}/${param}`
+   * */
+  defaultPath: boolean;
+}
+
+const stripped = (str: string) => str.replace(/[-/]/g, "");
+
+export class GuSSMParameter extends Construct implements IGrantable {
+  private readonly customResource: CustomResource;
+  readonly grantPrincipal: IPrincipal;
+
+  // eslint-disable-next-line custom-rules/valid-constructors -- I think stating an ID would be overkill for this, but happy to discuss
```

Comment:
I think there are plans to change this; see [similar a discussion](https://github.com/guardian/cdk/pull/326#discussion_r596057941) here. However perhaps `param` should be part of props, for consistency?

Reply:
Good shout, and done. It's actually becoming apparent that passing in an ID might solve a couple of issues down the line but will investigate.

The problem is essentially that it looks like we need to ensure we create new custom resources every time (not the lambda, but possibly everything else) in order to trigger the fetching of an SSM parameter every time. If you have a fixed ID, it _might_ be the case that it won't update past creation unless you somehow inform it that there's been a change.

Will investigate though.

Reply:
https://github.com/guardian/cdk/pull/352 revises the linting rules.